Makefile

```diff
@@ -14,7 +14,67 @@ build:
 	'$(VIRTUALENV_DIR)/bin/pip' install -r ./requirements.txt
 	'$(VIRTUALENV_DIR)/bin/python' setup.py install
 
+# Create a image with docker build & docker run.
+docker-build:
+	# e.g., docker build -t  <image_name> /path/to/Dockerfile/
+	# -t flag - to tag the image with a name so Docker recognizes this name.
+	docker build -t migrate2to3:v1 .
+
+docker-run: docker-build
+	# Runs docker-build part before docker-run part, i.e., docker-run DEPENDS on docker-build.
+	docker run -it migrate2to3:v1 bash
+
+# Create 2 virtualenvs. Test that Python 2.7 & Python 3.6 is running correctly
+# within each virtualenv respectively.
+# PYTHON 2.7 TESTS.
+test-help-tablesnap-py2.7: docker-build
+	docker run migrate2to3:v1 ./py27env/bin/python ./tablesnap --help
+
+test-help-tableslurp-py2.7: docker-build
+	docker run -e USER=root migrate2to3:v1 ./py27env/bin/python ./tableslurp --help
+
+test-help-tablechop-py2.7: docker-build
+	docker run migrate2to3:v1 ./py27env/bin/python ./tablechop --help
+
+test-help-tablesnap-sunset-py2.7: docker-build
+	docker run migrate2to3:v1 ./py27env/bin/python ./tablesnap-sunset --help
+
+test-help-tablesnap-sync-manifest-py2.7: docker-build
+	docker run migrate2to3:v1 ./py27env/bin/python ./tablesnap-sync-manifest --help
+
+# # PYTHON 3.6 TESTS.
+test-help-tablesnap-py3.6: docker-build
+	docker run migrate2to3:v1 ./py36env/bin/python ./tablesnap --help
+
+test-help-tableslurp-py3.6: docker-build
+	docker run -e USER=root migrate2to3:v1 ./py36env/bin/python ./tableslurp --help
+
+test-help-tablechop-py3.6: docker-build
+	docker run migrate2to3:v1 ./py36env/bin/python ./tablechop --help
+
+test-help-tablesnap-sunset-py3.6: docker-build
+	docker run migrate2to3:v1 ./py36env/bin/python ./tablesnap-sunset --help
+
+test-help-tablesnap-sync-manifest-py3.6: docker-build
+	docker run migrate2to3:v1 ./py36env/bin/python ./tablesnap-sync-manifest --help
+
+DIRECTORY=/tmp/tablesnap-test
+TIMEOUT_A=5
+TIMEOUT_B=10
+AWS_ACC_KEY_ID := $(shell aws configure get aws_access_key_id)
+AWS_SEC_ACC_KEY := $(shell aws configure get aws_secret_access_key)
+BUCKET=$(USER)-tablesnap-tests
+HARDLINKS_PATH=/tmp/tablesnap-hardlinks
+
+docker-test-tablesnap: docker-build
```

Comment:
I suggest renaming the `docker-test-tablesnap` target so it's more consistent with the other test targets above. Maybe `test-tablesnap-upload`?

Reply:
Made 2 rules to replace 1 old rule as we talked about and renamed to:
test-tablesnap-upload-py2.7
test-tablesnap-upload-py3.6

Reply:
New targets added, but I suggested changing their names in the comments below.